File: java/app/src/main/java/githubtododemo/githubclient/repos/item/WithOwnerItemRequestBuilder.java
package githubtododemo.githubclient.repos.item;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import githubtododemo.githubclient.repos.item.item.WithRepoItemRequestBuilder;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /repos/{owner}
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class WithOwnerItemRequestBuilder extends BaseRequestBuilder {
    /**
     * Gets an item from the GitHubTodoDemo.GitHubClient.repos.item.item collection
     * @param repo Unique identifier of the item
     * @return a WithRepoItemRequestBuilder
     */
    @jakarta.annotation.Nonnull
    public WithRepoItemRequestBuilder byRepo(@jakarta.annotation.Nonnull final String repo) {
        Objects.requireNonNull(repo);
        final HashMap<String, Object> urlTplParams = new HashMap<String, Object>(this.pathParameters);
        urlTplParams.put("repo", repo);
        return new WithRepoItemRequestBuilder(urlTplParams, requestAdapter);
    }
    /**
     * Instantiates a new WithOwnerItemRequestBuilder and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public WithOwnerItemRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/repos/{owner}", pathParameters);
    }
    /**
     * Instantiates a new WithOwnerItemRequestBuilder and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public WithOwnerItemRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/repos/{owner}", rawUrl);
    }
}

File: php/GitHub/Repos/Item/WithOwnerItemRequestBuilder.php
<?php

namespace Baywet\Githubtododemo\Github\Repos\Item;

use Baywet\Githubtododemo\Github\Repos\Item\Item\WithRepoItemRequestBuilder;
use Microsoft\Kiota\Abstractions\BaseRequestBuilder;
use Microsoft\Kiota\Abstractions\RequestAdapter;

/**
 * Builds and executes requests for operations under /repos/{owner}
*/
class WithOwnerItemRequestBuilder extends BaseRequestBuilder 
{
    /**
     * Gets an item from the baywet.githubtododemo.github.repos.item.item collection
     * @param string $repo Unique identifier of the item
     * @return WithRepoItemRequestBuilder
    */
    public function byRepo(string $repo): WithRepoItemRequestBuilder {
        $urlTplParams = $this->pathParameters;
        $urlTplParams['repo'] = $repo;
        return new WithRepoItemRequestBuilder($urlTplParams, $this->requestAdapter);
    }

    /**
     * Instantiates a new WithOwnerItemRequestBuilder and sets the default values.
     * @param array<string, mixed>|string $pathParametersOrRawUrl Path parameters for the request or a String representing the raw URL.
     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
    */
    public function __construct($pathParametersOrRawUrl, RequestAdapter $requestAdapter) {
        parent::__construct($requestAdapter, [], '{+baseurl}/repos/{owner}');
        if (is_array($pathParametersOrRawUrl)) {
            $this->pathParameters = $pathParametersOrRawUrl;
        } else {
            $this->pathParameters = ['request-raw-url' => $pathParametersOrRawUrl];
        }
    }

}
